R/license_note.R

```diff
@@ -0,0 +1,88 @@
+#' Generate LICENSE.note file.
+#'
+#' LICENSE.note generated by this function lists dependent Rust packages infomation.
+#' To use this function, the [cargo-lincense](https://crates.io/crates/cargo-license) command must be installed.
+#' @param force Logical indicating whether to re-generating LICENSE.note if LICENSE.note is already exists.
+#' @inheritParams register_extendr
+#' @return No return value, called for side effects.
+#' @export
+write_license_note <- function(path = ".", force = TRUE) {
+  if (!requireNamespace("RcppTOML", quietly = TRUE)) {
+    cli::cli_abort(
+      "The {.pkg RcppTOML} package is required to run the {.fun write_license_note} function.",
+      class = "rextendr_error"
+    )
+  }
+
+  if (!cargo_command_available(c("license", "--help"))) {
+    cli::cli_abort(
+      c(
+        "The {.code cargo license} command is required to run the {.fun write_license_note} function.",
+        "Please install cargo-license {.url https://crates.io/crates/cargo-license} first."
+      ),
+      class = "rextendr_error"
+    )
+  }
+
+  manifest_file <- rprojroot::find_package_root_file("src", "rust", "Cargo.toml", path = path)
+  out_file <- rprojroot::find_package_root_file("LICENSE.note", path = path)
+
+  if (!isTRUE(force) && file.exists(out_file)) {
+    cli::cli_abort(
+      c(
+        "LICENSE.note already exists.",
+        "If you want to re-generate LICENSE.note, set `force = TRUE` to {.fun write_license_note}."
+      ),
+      class = "rextendr_error"
+    )
+  }
+
+  list_license <- processx::run(
+    "cargo",
+    c(
+      "license",
+      "--authors",
+      "--json",
+      "--avoid-build-deps",
+      "--avoid-dev-deps",
+      "--manifest-path", manifest_file
+    )
+  )$stdout %>%
```

Comment:
Add some error handling if this fails for some reason?

Reply:
I think it is sufficient of the error indication of processx.